### ISSUE TITLE
Eslint warns about floating promises for `reply.header()`

### ISSUE THREAD

**Issue by rburgst** [fastify/fastify#4246]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.5.3

### Node.js version

16.13.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.5.1

### Description

When using eslint (with `@typescript-eslint/no-floating-promises`) + typescript it appears that

`reply.header()` is an async function that needs to be awaited 

However, when you actually `await reply.header` then the server hangs when running in jest.

### Steps to Reproduce

see https://github.com/rburgst/fastify-eslint-ts-problem

1. `npm install`
2. `npm test` -> all ok
3. `npm run lint` -> shows
    ```
      11:5  error  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
      ```
4. "fix" the lint error by changing `server.ts` to 
    ```typescript
    server.addHook('onSend', async (request, reply, payload) => {
        await reply.header('x-foo', 'bar')
        return payload
    })
    ```
    i.e. add `await` to `reply.header(...)

5. run `npm run test` -> now the test hangs and times out after 5s

### Expected Behavior

the types should make it clear that `reply.header` is a synchronous method

**Comment by mcollina** [comment]
TLDR the floating promises rule is incorrect. reply.header() does not return a promise but a Thenable, i.e. the reply object. This object can be awaited or not and unfortunately there is no way for us to tell tslint this rule does not apply to this object.

**Comment by mcollina** [comment]
cc @fastify/typescript

**Comment by Uzlopak** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-floating-promises.ts

Its isPromiseLike does check if the returned object is a thenable. So it is not fixable...

**Comment by rburgst** [comment]
FYI: using `await reply.header()` did not block the server in jest in fastify 3
